#### R scripts for Ctenophore Cydippid regeneration RNA-seq differential gene expression analysis

NOTE: run scripts in the following order. Some data files needed by scripts are generated by earliers scripts.

#### NOISeq_GO_FINAL.R

Pair-wise differential gene expression with NOISeq v2.44.0.  Must import transcript quantification files under /05-SUPP_DATA/Supp_Data2/RSEMcounts_B2_Trm_TPM_Supp.csv. Must also import Blast annotations for all ML gene models from 05-SUPP_DATA/Supp_Data3/HumRef2020ML_eval.txt  This script generates "NOISeq_sig_log2FC_v4.csv" which can also be found here: 05-SUPP_DATA/Supp_Data4/NOISeq_sig_log2FC_v4_Supp.csv file. This script also generates a list of DEG from NOISeq across all time points,without any duplicates that can be used in the comparison between the three methods( generate the consensus DEG list). This file is generated as "NOISeq_sig_v4.csv" in this script but can also be found here: 05-SUPP_DATA/Supp_Data5/NOISeq_sig_v4_Supp.csv. ALso contains the gene ontology analysis for the NOISeq DEG. Generates the redundance reduced GO terms for each time interval in each direction ( up or down regulated) called "NOIS_GO_redunred_v4.csv". This file can also be found here :05-SUPP_DATA/Supp_Data6/NOIS_GO_RedunRed_v4_Supp.csv Must import "ML2.2.aa_goterms_2.txt" -- #Mnemiopsis genes and their GO annotations generated from interproscan in Babonis et al 2018. 

#### EdgeR_FINAL.R

Pair-wise differential gene expression with edgeR v3.42.4. Must import transcript quantification files under /05-SUPP_DATA/Supp_Data2/RSEMcounts_B2_Trm_TPM_Supp.csv. Must also import Blast annotations for all ML gene models from 05-SUPP_DATA/Supp_Data3/HumRef2020ML_eval.txt  Generates "EdgeR_sig_pval_logFC_v4.csv" , which can also be found here: 05-SUPP_DATA/Supp_Data4/EdgeR_sig_pval_logFC_v4_Supp.csv.  This script also generates a list of DEG from edgeR across all time points,without any duplicates that can be used in the comparison between the three methods( generate the consensus DEG list). This file is generated as "EdgeR_sig_v4.csv" in this script but can also be found here: 05-SUPP_DATA/Supp_Data5/EdgeR_sig_v4_Supp.csv


#### EBSeqHMM_FINAL.R

Time-course differential gene expression with EBseqHMM v1.34.0.Must import transcript quantification files under /05-SUPP_DATA/Supp_Data2/RSEMcounts_B2_Trm_TPM_Supp.csv. Must also import Blast annotations for all ML gene models from 05-SUPP_DATA/Supp_Data3/HumRef2020ML_eval.txt Generates "EBseqHMM_sig_v4.csv" file -- contains all DEG across entire time course. This file can also be found here:  05-SUPP_DATA/Supp_Data5/EBseqHMM_sig_v4_Supp.csv. Generates "EBSeqHMM_blast_PP0.5_MLP_order.csv" -- DEG in each path with a posterior probaility cutoff of 0.5 with corresponding best BLAST hit. This file can also be found here: 05-SUPP_DATA/Supp_Data4/EBSeqHMM_blast_PP0.5_MLP_order_Supp.csv 

#### NOIS_EdgeR_mirror_FINAL.R

Quatification of DEG in each time interval in each direction (up or down regulated) for both pairwise programs ( edgeR and NOISeq).  Must also import Blast annotations for all ML gene models from 05-SUPP_DATA/Supp_Data3/HumRef2020ML_eval.txt  

#### consensus_venn_blastp_FINAL.R

Venn diagram of DEG found in EBseqHMM, edgeR, and NOISeq 
Must import 3 DEG lists generated from "EBSeqHMM_FINAL.R","EdgeR_FINAL.R"", and "NOISeq_GO_FINAL.R" or download from 05-SUPP_DATA/Supp_Data5


#### Heatmap_clust_tubulins_FINAL.R

Heatmap of gene expression in the consensus DEG.  Cluster analysis from the generated heatmap.  Must import "consensus_DE_v4.csv" which is the list of DEG found across all three methods generated from consensus_venn_blastp_FINAL.R.  Multi line graphs of Cluster.   Must import Blast annotations with "HumRef2020ML_eval.txt. Multi line graph of the expression of all of the alpha tubulin genes found in the Mnemiopisis gene models. 

#### RSEM_TPM_BOXANDWHISKER_V3.R

Box and whisker plots showing temporal gene expression of individually selected genes. Must import transcript quantification files under /05-SUPP_DATA/Supp_Data2/RSEMcounts_B2_Trm_TPM_Supp.csv. Must also import Blast annotations for all ML gene models from 05-SUPP_DATA/Supp_Data3/HumRef2020ML_eval.txt



